### PR TITLE
Feat/notifications: added pages for notifications

### PR DESCRIPTION
**Pull request by amandadupell**
#### What does this PR do?

Adds the main [Notifications](https://deploy-preview-1770--keen-mayer-a86c8b.netlify.app/components/notifications) page, the [Toast Notifications](https://deploy-preview-1770--keen-mayer-a86c8b.netlify.app/components/toast-notifications) subpage, and the [Status Indicator](https://deploy-preview-1770--keen-mayer-a86c8b.netlify.app/components/status-indicator) page

#### Where should the reviewer start?

Go into the components sections and explore all 3 new pages.

#### What testing has been done on this PR?

Visual testing locally.

#### How should this be manually tested?

Run the app locally.

#### Should this PR be mentioned in Design System updates?

Yes

### CODE REVIEW

**Comment by halocline** [comment]
@amandadupell - FYI, my 👀  are on this and I have some review comments in progress, but may not have them wrapped up by end of day.

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:17]: Can we link to the compliance guidance here?

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:34]: It might be nice to label each shape with the associated status

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:45]: can we bold the "Do This" / "Don't Do This" sections?

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:46]:
```suggestion
- When constructing content be sure to use words, phrases, and concepts familiar to the user, rather than internal jargon. 
```
Since "Content" isn't an actual component, would it make more sense for it just to be lowercase?

File: aries-site/src/pages/components/notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/notifications.mdx:12]: Might be helpful to bold these for a little extra contrast
```suggestion
- **Type** - In the broadest terms, what the notification is about
```

File: aries-site/src/pages/components/notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/notifications.mdx:21]: Maybe we could implement this as a markdown table so it looks more like the tables on the Voice and Tone page. Might also cleanup the code for this page a little. Documentation: https://www.markdownguide.org/extended-syntax/#tables

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:28]:
```suggestion
- Singular status updates
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:77]: Typically, we try to avoid using all caps and just stick with sentence case.

```suggestion
For best practice, keep the amount of information to three lines or under. This three line limit either can be a combination of both the title and the message or a three-lined title. If the information is heavy and long, consider using a different type of notification other than toast. Never truncate text within a toast. The main point is to keep the information concise and clear.
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:92]: Agreed with this point. However, often times, page level actions live at the top right of our layouts. Will this conflict? Or, where should toasts go and how do we know what kind of layout a user is going to be on when a toast appears?

File: aries-site/src/data/structures/components.js
Comment (halocline) [inline aries-site/src/data/structures/components.js:440]:
```suggestion
    description: 'Quickly provide peace-of-mind or call attention to items requiring a user's action.',
```

File: aries-site/src/data/structures/components.js
Comment (halocline) [inline aries-site/src/data/structures/components.js:445]:
```suggestion
    seoDescription: 'Highlight notification messages and alerts which require a user's attention. Status indicators provide peace-of-mind when all is well or call attention to items when a user needs to take action',
```

File: aries-site/src/data/structures/components.js
Comment (halocline) [reply aries-site/src/data/structures/components.js:440]: Are status indicators solely for use with notifications and alerts? Can they be used in tables, lists, etc.? If more broadly, perhaps we can focus on the quickly communicate detail?

File: aries-site/src/data/structures/components.js
Comment (halocline) [inline aries-site/src/data/structures/components.js:358]: Wondering if "Notification" should be under components and "Notifications" and "Status Indicators" under templates? Notifications being why and where these patterns applied within an application and then would reference "Notification" component page for the various flavors.

This is a bit of wondering aloud for discussion purposes. No changes for this in this PR.

**Review by halocline — COMMENTED** [review]
dropping comment incrementally

File: aries-site/src/examples/cardPreviews/toast-notification.js
Comment (halocline) [inline aries-site/src/examples/cardPreviews/toast-notification.js:38]: Eliminate double focus / button within a button:
Screen Shot 2021-08-13 at 2 59 21 PM


```suggestion
            <Button
              icon={<FormClose color={{ light: 'black', dark: 'white' }} />}
              plain
              tabIndex={-1}
            />
```

File: aries-site/src/examples/components/toast-notifications/ToastNotification.js
Comment (halocline) [inline aries-site/src/examples/components/toast-notifications/ToastNotification.js:8]: Keep name of export and file name consistent? Style/convention preference.

```suggestion
const ToastNotification = ({ toast, title, message, onClose }) => {
```

Comment (halocline) [reply aries-site/src/examples/components/toast-notifications/ToastNotification.js:8]: Or alternatively this file could be called `Notification.js`. Presumably this file becomes obsolete once Notification is available by Grommet.

File: aries-site/src/examples/components/toast-notifications/ToastNotification.js
Comment (halocline) [inline aries-site/src/examples/components/toast-notifications/ToastNotification.js:39]:
```suggestion
          <>
              <Text 
                  weight="bold" 
                  color={{ light: 'black', dark: 'white' }}
              >{title}</Text>
            {message && (
              <Text 
                  color={{ light: 'black', dark: 'white' }}
              >{message}</Text>
            )}
          </>
```

File: aries-site/src/examples/cardPreviews/toast-notification.js
Comment (halocline) [inline aries-site/src/examples/cardPreviews/toast-notification.js:8]: Will this get replaced with a `<Notification toast />` component once available from Grommet? If so, perhaps adding a TODO comment "replace this with ... when Notification component is available in Grommet" and/or add an issue to track this?

Comment (halocline) [reply aries-site/src/examples/cardPreviews/toast-notification.js:8]: Could you import Notification from `toast-notifications/ToastNotification.js` in order to re-use code? Looks very similar.

Basically thinking that if we are mimicking the implementation of the Notification component until it is available, can we mimic it in a single place, have all current imports point to it? Then when Notification is available, we simply replace the import from the local file to the import from Grommet?

File: aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js
Comment (halocline) [inline aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js:104]: Nice! 👏

File: aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js
Comment (halocline) [inline aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js:79]: Style/convention comment. Move to its own contained file and leave ToastNotificationExample only showing the example?

Also, the intent of the examples are for developers to be able to copy/paste the code example into their project. The toast diagram piece here would be extraneous, correct?

File: aries-site/src/pages/components/notifications.mdx
Comment (halocline) [inline aries-site/src/pages/components/notifications.mdx:10]:
```suggestion
To standardize language for Notifications across HPE projects, be sure to familiarize yourself with this taxonomy:
```

File: aries-site/src/pages/components/notifications.mdx
Comment (halocline) [inline aries-site/src/pages/components/notifications.mdx:105]: Is this Box needed?
```suggestion
<StateVsStatusDiagram />
```

File: aries-site/src/pages/components/status-indicator.mdx
Comment (halocline) [reply aries-site/src/pages/components/status-indicator.mdx:46]: Is there a relevant place to link within the Voice & Tone page for this?

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (halocline) [inline aries-site/src/pages/components/toast-notifications.mdx:45]: Can we eliminate Box, add pad to ToastDiagram and ...rest in ToastDiagram implementation?

File: aries-site/src/data/structures/components.js
Comment (amandadupell) [reply aries-site/src/data/structures/components.js:440]: yes these could be used outside of the context of notifications so the recommended change of language makes sense to me!

File: aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js
Comment (amandadupell) [reply aries-site/src/examples/components/toast-notifications/ToastNotificationExample.js:79]: makes sense to me!

File: aries-site/src/examples/components/toast-notifications/ToastNotification.js
Comment (amandadupell) [reply aries-site/src/examples/components/toast-notifications/ToastNotification.js]: moved to be in the notifications folder in `src/examples/components/notifications` and renamed the file to be `Notification.js`

File: aries-site/src/pages/components/notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/notifications.mdx:4]: Since this comes directly after the h1, it should be an h2
```suggestion
## What are they for?
```

File: aries-site/src/examples/components/notifications/TaxonomyTable.js
Comment (taysea) [inline aries-site/src/examples/components/notifications/TaxonomyTable.js:100]:
```suggestion
        header: <Text color="text-strong" weight="bold">Type</Text>,
```
Can we add color="text-strong" to all these table headings to align with way we style datatable headings?

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:33]: Can we remove the comment around "two sets of icons" since it seems we're just using one set?

File: aries-site/src/pages/components/status-indicator.mdx
Comment (taysea) [inline aries-site/src/pages/components/status-indicator.mdx:37]:
```suggestion
*** Remember that WCAG 2.1 guidelines states that at least three out of four elements must be present, therefore, this set will not pass accessibility requirements if it is not paired with another element such as well-written and relevant content. 
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:59]: Can we do similar thing here to what you did on the other page with the labels on the icons?

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:65]: Maybe rephrase so that link title is more informative
```suggestion
You can read more guidance about [status indicators](/components/status-indicator).
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:71]:
```suggestion
**1. Amount of Information**
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:77]:
```suggestion
**2. Interactive Content**
```

File: aries-site/src/pages/components/toast-notifications.mdx
Comment (taysea) [inline aries-site/src/pages/components/toast-notifications.mdx:79]:
```suggestion
To remain compliant with WCAG 2.1 guidelines, avoid placing interactive elements, like links and buttons, inside of a toast. These elements may be difficult to access with a screen reader and other assistive technology, so it is best to avoid complex toasts.
```

**Comment by britt6612** [comment]
LGTM! I also think we should all get together and decide where this should live before merging. second this 👆🏻

**Comment by britt6612** [comment]
After a discussion about the Notifications page:

- [ ] Move `status Indicators` to live under templates 
- [ ] Move `Toast Notifications` to live under templates

Make the following changes to the Notifications page:

- [ ] Change title to be singular so `Notification`
- [ ] Add Variants to Notification example and link to templates (put this before roadmap)
- [ ] Move `Notification Systems` up above variants

**Comment by britt6612** [comment]
LGTM once the merge conflict is fixed!

File: aries-site/src/examples/components/index.js
Comment (taysea) [inline aries-site/src/examples/components/index.js:40]: should we move these to the templates folder since that is where "Toast Notifications" page is?

File: aries-site/src/examples/components/notifications/StateVsStatusDiagram.js
Comment (taysea) [inline aries-site/src/examples/components/notifications/StateVsStatusDiagram.js:5]:
```suggestion
const color = 'text-strong';
```

File: aries-site/src/examples/components/toast-notifications/ToastDiagram.js
Comment (taysea) [inline aries-site/src/examples/components/toast-notifications/ToastDiagram.js:5]:
```suggestion
const color = 'text-strong';
```

File: aries-site/src/pages/components/notification.mdx
Comment (taysea) [inline aries-site/src/pages/components/notification.mdx:52]: If we update the folder where the examples live, we should also make sure to update the code links

**Review by taysea — COMMENTED** [review]
Tiny internal code structure comments, but otherwise looks good!

**Review by taysea — APPROVED** [review]
LGTM! Thanks for being patient with all the feedback.

**Review by halocline — APPROVED** [review]
Great job pushing this through.

**Pull request merged**
